cocoa/src/toga_cocoa/window.py

```diff
@@ -43,12 +44,69 @@ def windowShouldClose_(self, notification) -> bool:
         self.interface.on_close()
         return False
 
+    @objc_method
+    def windowWillClose_(self, notification) -> None:
+        # Setting the toolbar delegate to None doesn't disconnect
+        # the delegates and still triggers the delegate events.
+        # Hence, simply remove the toolbar instead.
+        if self.toolbar:
+            self.toolbar = None
+        # Disconnect the window delegate.
+        self.delegate = None
+
+        # Disconnecting the window delegate doesn't prevent custom
+        # methods like enteredMiniaturize_(which are performed with a
+        # delay i.e., having a delay != 0), from being triggered.
+        # Hence, check guards needs to be used in such custom methods.
```

Comment:
I guess you are correct. But now, since there are no window delegate methods that trigger with delay, so we can revisit this in the future, if needed.